File: benchmark/benchmark.cpp
#include "tinysegmenter.hpp"

#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include "tinysegmenter.hpp"

using namespace std;

int main() {
    ifstream ifs("timemachineu8j.txt");
    string str((istreambuf_iterator<char>(ifs)), istreambuf_iterator<char>());
    TinySegmenter segmenter;

    time_t start = time(NULL);
    for (int i = 0; i < 100; i++) {
        segmenter.segment(str);
    }
    time_t end = time(NULL);
    cout << (end - start);
}

File: benchmark/benchmark.js
var fs = require('fs');
var contents = fs.readFileSync('./timemachineu8j.txt');
var texts = contents.toString();
var TinySegmenter = require('./tinysegmenter');
var segmenter = new TinySegmenter.TinySegmenter();

var i = 0;
var start = Date.now();
for (i = 0; i < 100; i++) {
    segmenter.segment(texts);
}
var end = Date.now();
console.log((end - start)/1000 + "sec");
